src/util/timing/goblint_timing.ml

```diff
@@ -15,8 +15,8 @@ let next_tef_pid = ref 0
 
 module Make (Name: Name): S =
 struct
-  let enabled = ref false
-  let options = ref dummy_options
+  let enabled_dls = Domain.DLS.new_key (fun () -> false)
+  let options_dls = Domain.DLS.new_key (fun () -> dummy_options)
```

Comment:
Is it necessary for these to be domain-local? These are more like global options than state.
#1550 mentions that maybe only `current` needs to be domain-local.

Reply:
Felix himself has introduced all of them in https://github.com/goblint/analyzer/commit/62422e9a14dedbb913c5fabc2bdb2cd2fa37c1b0 

As far as can see `enabled_dls` is rather tracking if it is active in a given moment (set via `start` and `stop`), and making it a reference makes the solver crash at the end. Options however are global options and can be made into refs.

9e1614d4f9b93b809d135a8d48671d1f367500b3 reflects this now.

Reply:
> As far as can see `enabled_dls` is rather tracking if it is active in a given moment (set via start and stop), and making it a reference makes the solver crash at the end.

But we only ever call `start` at the very beginning of Goblint and never even call `stop`. So I don't understand why there would be any issue. `start` also modifies `options` and that's fine.

Reply:
Interesting, making them into refs did cause issues, so I will need to investigate this further.

Reply:
When `enabled` is a regular ref, enabling timing results in `Fatal error: exception Stdlib.Stack.Empty` after printing the timings, so far I could not figure out why. Maybe you can give me some pointers at gobcon?

Reply:
Oh, maybe `enabled` needs to be `Atomic` to avoid some race condition and out-of-date values between domains?

Reply:
With some help from @michael-schwarz we did figure out what went wrong.

The enter method requires a root frame, which is created only once at start. Since the stack is in DLS, every thread should create a root frame, which is now being handled in `split_from_parent` in DLS. 

The values recorded by the threads are not included in the final report, but should be correct.